File: examples/Coverage1.py
#!/usr/bin/env python3

# pylint: disable = bad-whitespace
# pylint: disable = invalid-name
# pylint: disable = missing-docstring
# pylint: disable = pointless-string-statement

# ------------
# Coverage1.py
# ------------

# https://coverage.readthedocs.org

from unittest import main, TestCase

def cycle_length (n: int) -> int :
    assert n > 0
    c = 1
    while n > 1 :
        if (n % 2) == 0 :
            n = (n // 2)
        else :
            n = (3 * n) + 1
        c += 1
    assert c > 0
    return c

class MyUnitTests (TestCase) :
    def test (self) :
        self.assertEqual(cycle_length(1), 1)

if __name__ == "__main__" : # pragma: no cover
    main()

""" #pragma: no cover
% which coverage
/usr/local/bin/coverage



% coverage --version
Coverage.py, version 4.4.2 with C extension
Documentation at https://coverage.readthedocs.io



% coverage --help
...



% coverage run --branch Coverage1.py
.
----------------------------------------------------------------------
Ran 1 test in 0.000s

OK



% coverage report -m
Name           Stmts   Miss Branch BrPart  Cover   Missing
----------------------------------------------------------
Coverage1.py      16      4      4      1    61%   20-24, 19->20
"""

File: examples/Coverage1.js
#!/usr/bin/env node

/* global
    suite,
    test
*/

// ------------
// Coverage1.js
// ------------

// https://gotwarlost.github.io/istanbul/

"use strict";

var assert = require('assert');

function cycle_length (n) {
    assert(n > 0);
    var c = 1;
    while (n > 1) {
        if ((n % 2) === 0) {
            n = n / 2;}
        else {
            n = (3 * n) + 1;}
        c += 1;}
    assert(c > 0);
    return c;}

suite('cycle_length',
    function () {
        test('test',
            function () {
                assert.equal(cycle_length(1), 1);});});

/*
% which istanbul
/usr/local/bin/istanbul



% istanbul help
...
istanbul version:0.4.5



% istanbul cover _mocha -- -u tdd Coverage1.js
  cycle_length
    ✓ test


  1 passing (4ms)

=============================================================================
Writing coverage object [/Users/downing/Dropbox/examples/javascript/coverage/coverage.json]
Writing coverage reports at [/Users/downing/Dropbox/examples/javascript/coverage]
=============================================================================

=============================== Coverage summary ===============================
Statements   : 71.43% ( 10/14 )
Branches     : 0% ( 0/2 )
Functions    : 100% ( 3/3 )
Lines        : 71.43% ( 10/14 )
================================================================================
*/
